src/datasets/datasets.controller.ts

```diff
@@ -210,18 +240,57 @@ export class DatasetsController {
       // NOTE: We need DatasetClass instance because casl module can not recognize the type from dataset mongo database model. If other fields are needed can be added later.
       const datasetInstance = new DatasetClass();
       datasetInstance._id = "";
-      datasetInstance.pid = "";
+      datasetInstance.pid = dataset.pid || "";
       datasetInstance.accessGroups = dataset.accessGroups || [];
       datasetInstance.ownerGroup = dataset.ownerGroup;
       datasetInstance.sharedWith = dataset.sharedWith;
       datasetInstance.isPublished = dataset.isPublished || false;
       datasetInstance.owner = dataset.owner;
       datasetInstance.ownerEmail = dataset.ownerEmail;
       if (user) {
-        const ability = this.caslAbilityFactory.createForUser(user);
-        const canCreate = ability.can(Action.Create, datasetInstance);
-        if (!canCreate) {
-          throw new ForbiddenException("Unauthorized to create this dataset");
+        const { isPartOfAdminGroups, userCanCreateDatasetWithPid } =
+          this.getUserPermissionsFromGroups(user);
+        if (
+          datasetInstance.pid &&
+          !userCanCreateDatasetWithPid &&
+          !isPartOfAdminGroups
+        ) {
+          throw new ForbiddenException(
+            "Unauthorized to create datasets with explicit PID",
+          );
+        } else {
+          // NOTE: If it can create dataset but not part of ADMIN_GROUPS and not part of CREATE_DATASET_WITH_PID_GROUPS,
+          // then we make sure that pid is not provided by user but it is generated by the system.
+          if (!isPartOfAdminGroups && !userCanCreateDatasetWithPid) {
+            delete dataset.pid;
+          }
```

Comment:
Should we return an error if we are allowed to create a dataset but without pid?
I wonder if that is a better behavior than allow it and change the pid.